docs_src/design/ucr/0003-Extending-Device-Data.md

```diff
@@ -0,0 +1,106 @@
+The new analytic, utility, and north-bound microservices almost always need to add Device Resources to manage their configuration, transforms, and status reporting. These Resources are usually needed on a per-Device basis (rather than just overall service configuration or status), which can be seen as extending (adding to) the data of the original south-bound devices.
+
+Adding configuration and status via Devices that extend the original south-bound Device, and new Device Resources,
+make this configuration and status data easily accessible and translatable to other Application Services and to the UI via REST;
+we think that this general solution is better than disparate solutions which add custom APIs in each Application Service to Get and Set this data.
+
+What is needed is a common means of showing the relationship between these added Resources and their
+original south-bound Device; that is, to indicate that these Resources "extend" the original Device.
+
+It is desirable that the means of conveying this information become standardized for those EdgeX microservices
+which provide and use it, hence proposing here that there be a common EdgeX way defined to do this.
+
+### Target Users
+- Product Developers
+- Device Owner
+- Device User
+- Device Maintainer
+- Cloud User
+- Service Provider
+- Software Integrator
+
+### Description
+Picture the extremely simple case of a south-bound sensor device that just measures Temperature and Humidity and provides these as Device Resources. If we then add analytic and north-bound microservices:
+- A Trending service that has Device Resources to indicate that Temperature and Humidity are trended for, eg, Minimum, Average, and Maximum over a 1 hour trend interval.
+- An Alarming service that has Device Resources to describe the Alarm Rules used to monitor Temperature and Humidity, plus a device-level InAlarm status.
+- A Cloud service that reports not just the Temperature and Humidity but also their Trend configuration and Alarm Rule Resources. 
+In addition, the Cloud service adds its own Resources to direct the Cadence with which this Device's data is reported.
+
+Now scale this up to 100 such Temperature/Humidity sensors, and it grows difficult to match all of the added
+Resources to their original sensor data. And add the requirement that all these resources must be able to be seen 
+and managed locally via REST or Message Bus, and potentially from north-bound services like Modbus/TCP, and from 
+the Cloud (because everybody wants to control everything from the Cloud). 
+
+Furthermore, from the end user's point of view, the Trend configuration, Alarm Rules, and Cloud Cadence that are added for a given Device are all seen as aspects of the Temperature/Humidity Device, as is common 
+for Digital Twin representations, and not as separated, free-standing entities. 
+So there must be some means to relate the extended Device Resources to the original south-bound Device and its
+Device Resources.
```

Comment:
The Device Resources are define in the associated device profile, aka module. So we are really talk about extending this module that describe a class of devices.

Reply:
Yes. From the point of view of the external client connected to our north-bound services, it sees the "Watt-o-Meter" device with its energy measurements AND its trend, alarm, and cloud cadence configurations all as properties of the "Watt-o-Meter" device, not as a "Watt-o-Meter" device with its energy measurements and three "shadow" devices that happen to have trend, alarm, and cloud cadence configurations that are related (somehow) to that Watt-o-Meter device.